analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/NullPointerDereferenceBase.cs

```diff
@@ -20,34 +20,45 @@
 
 using SonarAnalyzer.SymbolicExecution.Constraints;
 
-namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
+
+public abstract class NullPointerDereferenceBase : SymbolicRuleCheck
 {
-    public abstract class NullPointerDereferenceBase : SymbolicRuleCheck
-    {
-        internal const string DiagnosticId = "S2259";
+    internal const string DiagnosticId = "S2259";
 
-        protected virtual bool IsSupressed(SyntaxNode node) => false;
+    protected virtual bool IsSupressed(SyntaxNode node) => false;
 
-        protected override ProgramState PreProcessSimple(SymbolicContext context)
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        if (ReferenceOrDefault(context.Operation.Instance) is { Syntax: var syntax} reference
```

Comment:
I don't think we need the `OrDefault` part. It should just describe what it primarily returns, the null-check expresses that it might not always be the case.

The syntax is also bit abusing the pattern matching, using `reference.Syntax` would be clearer to understand below
```suggestion
        if (Reference(context.Operation.Instance) is { } reference
```

Reply:
I don't like just omitting the `OrDefault` because it makes it way harder to understand that this method actually checks for something. It's also the way a lot of methods in the .NET class library are being called. I feel like there might be a name that's easy to understand and succinct. I'll try to think of something.

Reply:
Can you explain what exactly you dislike of this use of Pattern Matching? I find it is a good use case for it and have seen similar code in a lot of other places in the code base.

Reply:
To clarify the "this" in "this use", I don't like the `var` part. 
Having `Reference(...) is { } reference` is completely fine as it introduces a newly computed state.
I don't like The `var syntax` part, as its abusing the `var`  (that should be used only when it really makes sense) a little, and it makes reading the code harder. To read this, I need to track additional state `syntax` and think about the question where it came from. Comming to a conclusion that it's just a duplicate state with already exisiting `reference.Syntax`.